### DISCUSSION TITLE
Use folders on image interface

### DISCUSSION THREAD

**Discussion by Meldiron** [directus/directus#4999] [Feature Requests]
If we add `Image interface` there are 3 options: from disk, from gallery, from URL

![image](https://user-images.githubusercontent.com/19310830/114406805-1d1a3580-9ba8-11eb-9461-6b041f5440cb.png)

If we pick option `from mgallery`, we get list of all files uploaded to Directus that we can filter by name

![image](https://user-images.githubusercontent.com/19310830/114406867-2c00e800-9ba8-11eb-9ba0-231f3ea8c463.png)

My idea is, can we filter by folders? If I enter file library, there is the left panel to put files into folder in order to prevent mess

![image](https://user-images.githubusercontent.com/19310830/114406984-433fd580-9ba8-11eb-9904-230438acf142.png)

**Having a folder selection instead of list of all files would be awesome ♥**

**Comment by benhaynes** [comment]
Another way to accomplish this is to include navigable Folders in the list of files, but at the top of the list (see screenshot below). This would give all the functionality, without needing to add a sidebar to the drawer. Just an idea!

Duplicate of: https://github.com/directus/directus/discussions/3464

![Screen Shot 2021-04-12 at 2 04 20 PM](https://user-images.githubusercontent.com/522079/114440379-0bc92d00-9b98-11eb-8a28-5bba8125b25a.png)

**Reply by rijkvanzanten** [reply]
> without needing to add a sidebar to the drawer.

The nice thing of this approach is that it works regardless of the layout you have configured for files

**Reply by abertay-university** [reply]
@rijkvanzanten, to further this idea, could you include a filter option on the file/image interface where we can hide or limit the view of images? This would be very useful for large libraries.

As an example, if I create an icon selector and have a folder for all my icons, I would like the image interface to only show the contents of the icon folder. Also, if I am choosing an existing image, it would be useful to hide the icon folder.
{ "folder": { "_eq": "<folderID>" } }
{ "folder": { "_neq": "<folderID>" } }
By using the existing filter model, the options are endless

**Reply by rijkvanzanten** [reply]
Yes absolutely, I wanna do that across the board for every relational interface. As a matter of fact, there's already a PR in the works to get that in 🙂  https://github.com/directus/directus/pull/7080

**Reply by abertay-university** [reply]
Yay, @rijkvanzanten you're a star ⭐️

**Reply by Michael-Nussbaumer** [reply]
This would be also pretty nice to have in the file library aswell. I currently have a customer which is used to have a pretty huge amount of folders and files (+500 folders for customers with some organization with customer folders aswell) Here the drawer isn't wide enough to show the full name and it is pretty confusing without seeing the subfolders in the interface itself.

**Comment by rijkvanzanten** [comment]
This is going to be implemented. You can follow along on https://github.com/directus/directus/issues/15968
